src/Reddit/Reddit.ts

```diff
@@ -1,49 +1,327 @@
+      }
+    },
+    vote: {
+      btn: ".voteButton",
+      up: ".voteButton[aria-label=\"upvote\"]",
+      down: ".voteButton[aria-label=\"downvote\"]",
+      pressed: ".voteButton[aria-pressed=\"true\"]",
+      unpressed: ".voteButton[aria-pressed=\"false\"]",
+    }
+  }
+}
+
+function thingAtIndex(i: number): string {
+  if (isOldReddit) {
+    return `${ reddit.old.post.thing }[${ thingAttr }="${ i }"]`
+  } else {
+    return `${ reddit.latest.post.thing }[${ thingAttr }="${ i }"]`
+  }
 }
 
 function clickIfExists(selector: string) {
-  const el = document.querySelector<HTMLElement>(selector);
-  if (el) el.click();
+  const el = select<HTMLElement>(selector)
+  if (el) el.click()
+}
+
+function clickIfDisplayed(el: HTMLElement) {
+  if (parseFloat(getComputedStyle(el).width)) {
+    el.click()
+  }
+}
+
+function genPostNumberElement(number): HTMLElement {
+  const span = document.createElement("span")
+  span.textContent = number
+  span.className = "post-number"
+
+  setStyles({
+    position: "absolute",
+    bottom: "2px",
+    right: "2px",
+    fontWeight: 700,
+    opacity: .3
+  }, span)
+
+  return span
+}
+
+function addRedditAPostsAttributes(posts: NodeListOf<HTMLElement>, isOld: boolean) {
+  if (isOld) {
+    return posts.forEach((el) => {
+      index += 1
+      el.setAttribute(thingAttr, `${ index }`)
+      const rank = select<HTMLElement>(".rank", el)
+
+      setStyles({
+        display: "block",
+        marginRight: "10px",
+        opacity: "1 !important"
+      }, rank!)
+    })
+  }
+
+  posts.forEach(setAttributes)
+}
+
+function setAttributes(post: HTMLElement) {
+  if (
+    post
+    && getComputedStyle(post).display !== "none"
+    && !post!.querySelector(".post-number")
+  ) {
+    index += 1
+
+    post.setAttribute(thingAttr, `${ index }`)
+    post.style.position = "relative"
+    post.appendChild(genPostNumberElement(index))
+  }
+}
+
+function observerCallback(mutationList) {
+  const { old, latest } = reddit
+  const postSelector = isOldReddit ? old.post.thing : latest.post.thing
+
+  mutationList.forEach(it => {
+    it.addedNodes.forEach(node => {
+      const post = select<HTMLElement>(postSelector, node)
+      setAttributes(post!)
+    })
+  })
+}
+
+function createObserver(el: Element) {
+  observer = new MutationObserver(observerCallback)
+  observer.observe(el!, { childList: true })
+}
+
+function setParentContainer(posts: NodeListOf<HTMLElement>): Maybe<ParentNode> {
+  return posts![0].parentNode!.parentNode!.parentNode
+}
+
+function onLoad() {
+  if (isDOMLoaded) return
+
+  const { old, latest } = reddit
+  const postSelector = isOldReddit ? old.post.thing : latest.post.thing
+
+  posts = selectAll<HTMLElement>(postSelector)
+
+  isDOMLoaded = true
+
+  addRedditAPostsAttributes(posts, isOldReddit)
+
+  if (!isOldReddit) {
+    scrollContainer = setParentContainer(posts)
+    createObserver(scrollContainer! as Element)
+  }
+}
+
+function onPopState() {
+  setTimeout(() => {
```

Comment:
I know we discussed why there needs to be a timeout here, but you should put it in a comment so future devs know